### ISSUE TITLE
About the two-stage training procedure.

### ISSUE THREAD

**Issue by kxgong** [bytedance/1d-tokenizer#15]
Hi, dear authors, thanks for your wonderful work.

I still have some questions about the two-stage training pipeline. First, I want to claim that I totally understand the TiTok tokenization process shown in Fig. 3.

In the warm up, how to leverage maskgit-vqgan's code for training. I believe the code number of maskgit-vqgan will be 256 while the settings of TiTok will range from 32 ~ 128. This is a mismatch. 

Do the mask tokens for decoding only start to train during the second stage?

I apologize for missing and misunderstanding any details in the paper to address my questions.

**Comment by cornettoyu** [comment]
Hi,

Thank you for your interests in our work and patience. We have released the two-stage training code and feel free to check it and hope it addresses your question.
